src/include/schema.h

```diff
@@ -121,6 +121,23 @@ struct __wt_table {
 	    &S2C(session)->checkpoint_lock, WT_SESSION_LOCKED_CHECKPOINT, op)
 
 /*
+ * WT_WITH_CHECKPOINT_LOCK_NOWAIT --
+ *	Return error if unable to acquire the checkpoint lock,
+ *	perform an operation if lock acquired, drop the lock.
+ */
+#define	WT_WITH_CHECKPOINT_LOCK_NOWAIT(session, ret, op) do {		\
```

Comment:
Is there a way to use the generic `WT_WITH_LOCK` that already has support for trylock by setting the `WT_SESSION_LOCK_NO_WAIT` flag instead for this function?  This macro seems far more specific than all the other locking macros here.